### DISCUSSION TITLE
@mdx-js/language-server throws at initialisation with 'TypeScript module is missing'

### DISCUSSION THREAD

**Discussion by GNRSN** [mdx-js/mdx#2427] [Q&A] [answered]
I'm trying to set up mdx_analyzer as an LSP server in Neovim but the serves crashes on initialisation

Using:
@mdx-js/language-server: 0.2.2 (latest)
on node 18.18.0

The error thrown, 'TypeScript module is missing', stems [from this assertion](https://github.com/mdx-js/mdx-analyzer/blob/b50897668277e1c354d90159f6b902c08457493c/packages/language-server/lib/language-server-plugin.js#L39) From neovim's side, [communication with the language server happens through the cli with `--stdio`
](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md#mdx_analyzer)
See 

I've done some sanity checkes like ensuring I have a clean npm install but to no avail. 

I'm confused about installation section in the readme referring to `remark-rehype` so I'm unsure if I'm missing some component of setup? I couldn't find a Typescript dependency in the source so I my current theory is that I miss a typescript lib reference, [similar to the path option for tsserver?](https://github.com/typescript-language-server/typescript-language-server/blob/master/docs/configuration.md#tsserver-options) but I'm unsure how to specify this or if volar expects this to be installed somehere in relation to cwd for execution?

Any advice for what I may be missing in terms of configuration or on how to further debug this will be much appreciated!

**Comment by remcohaszing** [comment, marked as answer]
Hey @GNRSN, it’s great to see someone setup the MDX language server in another editor.

Admittedly, the docs need some work. The mention of `remark-rehype` is a copy-paste error. You don’t need `remark-rehype` at all.

The language server requires you to pass the path to the TypeScript library via initialization options. You can see how it’s done in the VSCode extension [here](https://github.com/mdx-js/mdx-analyzer/blob/main/packages/vscode-mdx/src/extension.js#L48-L50). MDX analyzer is based on Volar, the same technology that powers editor integration for Astro and Vue. I suggest you have a look at https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/astro.lua and https://github.com/neovim/nvim-lspconfig/blob/master/lua/lspconfig/server_configurations/volar.lua.

**Reply by GNRSN** [reply]
Thank you! Just copying the Astro config was enough to get me up and running. Hopefully this is helpful to anyone else trying to setup mdx for Neovim 😄
